redbot/cogs/mutes/mutes.py

```diff
@@ -401,9 +401,12 @@ async def _handle_channel_unmutes(self):
                         log.debug(f"Creating task: {task_name}")
                         if task_name in self._unmute_tasks:
                             continue
-                        self._unmute_tasks[task_name] = asyncio.create_task(
-                            self._auto_channel_unmute_user(guild.get_channel(channel), mute_data)
-                        )
+                        if guild_channel := guild.get_channel(channel):
+                            self._unmute_tasks[task_name] = asyncio.create_task(
+                                self._auto_channel_unmute_user(guild_channel, mute_data)
+                            )
+                        else:
+                            await self.config.channel_from_id(channel).clear()
```

Comment:
I'm not sure we really want to remove channel mutes just because they aren't in our channel cache. I would probably just remove this part.